ext/mixed/js/template-handler.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020  Alex Yatskov <alex@foosoft.net>
+ * Author: Alex Yatskov <alex@foosoft.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+class TemplateHandler {
+    constructor(html) {
+        this._templates = new Map();
+        this._html = html;
+        this._doc = null;
+
+        this._initialize();
+    }
+
+    _initialize() {
+        this._doc = new DOMParser().parseFromString(this._html, 'text/html');
```

Comment:
I would just leave `this._doc` as a temp var `doc` since it's not used afterwards, and similar for `_html`. Everything could just be put into the constructor also, it doesn't really benefit much to abstract it into a private `_initialize` function which isn't `async` and is of very little complexity.